src/tests/ftest/cart/tests_common.h

```diff
@@ -498,4 +493,80 @@ tc_srv_start_basic(char *srv_group_name, crt_context_t *crt_ctx,
 	D_ASSERTF(rc == 0, "crt_group_size() failed; rc=%d\n", rc);
 }
 
+struct tc_log_msg_cb_resp {
+	sem_t	sem;
+	int	rc;
```

Comment:
this field is not really useful here; if cci_rc != 0 we exit early without setting it, so it will only be set for rc=0 case

Reply:
Half right.  Need to remove the early exit in the call back function.  Still need to free the semaphore and to relay back to the main function that a failure occurred.

Reply:
In general, how do we know that the response address in CB is still valid?  My address is a local structure on the stack.  If the semaphore times out, then the function exit before the call back is executed.  However, the rpc may complete afterwards and the call back is executed.

Reply:
In general to solve this problem when you encounter a timeout then you want to abort the rpc request and wait for it to complete before you destroy the semaphore. Once rpc is aborted it is supposed to (at some point, which should be rather quick) to invoke completion callback with -DER_CANCELED error code or aborted error code in cci_rc.